File: ApiProyectoFinal4/ProyectoFinal4.Modelos/Entidades/GastoViaje.cs
using System;

namespace ProyectoFinal4.Modelos.Entidades
{
    public class GastoViaje
    {
        public int GastoViajeId { get; set; }
        public int ViajeId { get; set; }
        public TipoGasto ClaveTipoGasto { get; set; }
        public double Monto { get; set; }

        public string DescripcionTipoGasto
        {
            get
            {
                switch (ClaveTipoGasto)
                {
                    case TipoGasto.Gasolina:
                        return "Gasolina";
                    case TipoGasto.Casetas:
                        return "Casetas";
                    case TipoGasto.Alimentos:
                        return "Alimentos";
                    case TipoGasto.Hospedaje:
                        return "Hospedaje";
                    case TipoGasto.Otros:
                        return "Otros";
                    default:
                        throw new Exception("Tipo de gasto desconocido");
                }
            }
        }
    }
}

File: TransportaAppAdmin/app/src/main/java/com/mtwdm/android/transportaappadmin/model/entidades/GastoViaje.java
package com.mtwdm.android.transportaappadmin.model.entidades;

public class GastoViaje {

    public int gastoViajeId;
    public int viajeId;
    public int claveTipoGasto;
    public double monto;

    public String getDescripcionTipoGasto() throws Exception {
        switch (claveTipoGasto) {
            case TipoGasto.GASOLINA:
                return "Gasolina";
            case TipoGasto.CASETAS:
                return "Casetas";
            case TipoGasto.ALIMENTOS:
                return "Alimentos";
            case TipoGasto.HOSPEDAJE:
                return "Hospedaje";
            case TipoGasto.OTROS:
                return "Otros";
            default:
                throw new Exception("Tipo de gasto desconocido");
        }
    }

}
